File: 2025-11-15/example_comparing_c_python/c/main.c
#include <stdio.h>

int main(int argc, char** argv)
{

    int array[10] = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10};

    {
        int total = 0;
        for (int index = 0; index < 10; index++)
        {
            total += array[index];
        }
        printf("Total (method 1) is: %d\n", total);
    }

    {
        int total = 0;
        for (int index = 0; index < 10; index++)
        {
            total = total + array[index];
        }
        printf("Total (method 2) is: %d\n", total);
    }

    {
        int index = 0;
        int total = 0;
        while(index < 10)
        {
            total += array[index];
            index++;
        }
        printf("Total (method 3) is: %d\n", total);
    }
}

File: 2025-11-15/example_comparing_c_python/python/main.py
from functools import reduce

if __name__ == "__main__":

    array = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

    # Basic for loop using the iterator
    total = 0
    for val in array:
        total += val
    print(f"Total (method 1) is {total}")

    # Basic for loop using indexing
    total = 0
    for index in range(0, len(array)):
        total += array[index]
    print(f"Total (method 2) is {total}")

    # Using the inbuilt 'sum' function
    print(f"Total (method 3) is {sum(array)}")

    # Using a functional method
    print(f"Total (method 4) is {reduce(lambda x, y: x + y, array)}")